Tests/Common/Extensions/DateExtensionTests.swift

```diff
@@ -59,4 +59,18 @@ class DateExtensionTest: UnitTest {
     func test_isOlderThan_givenDateThatIsNewer_expectFalse() {
         XCTAssertFalse(Date().subtract(10, .minute).isOlderThan(Date().subtract(60, .minute)))
     }
+
+    // MARK: formatToIso8601WithMilliseconds
+
+    func test_givenDate_formatToIso8601WithMilliseconds_expectString() {
+        let date = dateUtilStub.now
+        let formattedDate = date.string(format: .iso8601WithMilliseconds)
+
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
+        formatter.timeZone = TimeZone(identifier: "UTC")
+        let expectedDate = formatter.string(from: date)
```

Comment:
This test might accurately test the use case, but unless I fully understand this highlighted block of code, I cannot be certain. 

Also, is this highlighted block of code has a bug, then we cannot be sure if the code under test is working or not. 

As a suggestion, I suggest using a hard-coded string as the expected value. Something like: 

```swift
    func test_givenDate_formatToIso8601WithMilliseconds_expectString() {
        let givenDate = Date(...)
        let expectedString = "2023-12-04 ..."

        let actualString = givenDate.string(format: .iso8601WithMilliseconds)

        XCTAssertEqual(expectedString, actualString)
     }
```

Reply:
Updated https://github.com/customerio/customerio-ios/pull/490/commits/fa796e28a0e2e8150c301e4afce7af98a3f389c8